src/EndevorController.ts

```diff
@@ -212,7 +213,15 @@ export class EndevorController {
         this.updateIDs(connName);
       } else {
         const endevorDataProvider = new EndevorDataProvider();
-        endevorDataProvider.addSession(connName);
+        try {
+          endevorDataProvider.addSession(connName);
+        } catch (error) {
```

Comment:
Why you determine all possible errors here as a this specific issue error?

Is it be better to evolve the basic error from 'Profiles.loadNamedProfile' to more specific error type, and catch this type here. 

For example, the code from 'Profiles.loadNamedProfile' can be modified into smt like this:
`throw new ProfileNotFoundError('Could not find profile named: ' + name + '.')`. 
And this catch can be more specific for desired error type and more common for general errors. What do you think about it?

Reply:
The point of using logger is to allow the extension to activate. Currently, it is throwing an error and stopping the extension from activating because the profile that is saved in settings.json doesn't physically exist.

Reply:
Yeah, I see the point, and handling errors can be different for different error types. 

For example, if 'addSession' method throws an Error related to the wrong syntax or about something went wrong during the tree refreshing, which is not related to the profiles stuff - you will handle such errors with the message about profiles, and it can be confusing to developers and to users.

You can provide different catch flows for different error types here, so general errors will produce one message, but more specific ones - another (for example).

Reply:
So far this is the error that I am receiving. I can filter this maybe through the message itself.

![Untitled](https://user-images.githubusercontent.com/45042840/100628664-c0b92680-3328-11eb-80ec-d5cf7cdea64a.png)

Reply:
You can use more convenient mechanism of catching with types, like 'instance of' or checking the error type.

So the one possible solution here, is it create more specific error class, like:
`
export class InvalidUriError extends Error {
    constructor(message: string) {
        super(message);
        this.name = "InvalidUriError";
    }
}
`

Then, you can throw it in the source code, and catch it with more specific catch block:
`
catch(e) {
      if (e instanceof IncorrectUriError) {
        logger.error(e.message);
      } else {
        logger.error(`something went wrong: ${e.message}`);
      }
    }
`

There are just examples above.

What are the benefits for you? That you have controlled error handling flow. So if you have one error - it will indicate one situation and required one handling flow, but if you have another error type - you can do it in another way.

Reply:
I like this and I don't mind doing it this way. Maybe this approach could also be a new refactoring/enhancement issue? so we can also change how errors are handled in the codebase. what do you think?

Reply:
Oh, since these huge codebase changes (full rewriting the codebase) are about to come, I'm not sure, this story will be relevant after them

Reply:
In my opinion it is still good to open the issue as a placeholder so we won't forget.

Reply:
From this conversation, I am changing the error to: `Error: Could not find profile named: ${connName}. Please check your configuration in settings.json`

Is this text okay with people?

Reply:
Yep, good for me!

Reply:
The prefix of `Error: ` is extraneous. Also it doesn't match the icon used in the pop-up, which is ⚠️ (Warning).
I would rather have the following:

```
logger.warn(
            error.message,
            `Profile ${connName} defined in settings.json was not found. Make sure the profile is defined or remove it from settings.json. \n\n ${error.stack}`
          );
```
![image](https://user-images.githubusercontent.com/45846673/101661106-2ce20b80-3a48-11eb-99da-546aa9722381.png)

Reply:
Ok, adding this...